src/webui/abstractwebapplication.cpp

```diff
@@ -86,6 +87,18 @@ struct WebSession
     }
 };
 
+namespace
+{
+    QUrl urlFromHostHeader(const QString &hostHeader)
+    {
+        const QRegularExpression re {QLatin1String("\\w+://")};
+        if (hostHeader.indexOf(re) == 0)
```

Comment:
:\  The original code has no problem on linux, and this non-uniform behavior of `QUrl::fromUserInput` is making life harder... could it be a bug in Qt?

```c++
QUrl url = QUrl::fromUserInput("[::1]:8080");
// my results on windows:
// url.isValid() == true
// url.scheme() == "file"
// url.port() == -1
// url.host() == ""  how could it be empty!?

// my results on linux:
// url.isValid() == true
// url.scheme() == "http"
// url.port() == 8080
// url.host() == "::1"
```

Also I would write it as:
```c++
QUrl urlFromHostHeader(QString str)
{
  str = str.trimmed();
  if (str.startsWith("http://") || str.startsWith("https://"))
    return str;
  return (QLatin1String("http://") + str);
}
```

Reply:
>`url.host() == ""` how could it be empty!?

It's empty since file has no host, only path.
>Also I would write it as: ...

Since both `Host` and `X-Forwarded-Host` should not contain scheme it can be written the following way:
```c++
inline QUrl urlFromHostHeader(QString str)
{
  return QUrl(QLatin1String("http://") + str);
}
```

Reply:
>Since both Host and X-Forwarded-Host should not contain scheme it can be written the following way:

OK.

>>url.host() == "" how could it be empty!?
>
>It's empty since file has no host, only path.

`QUrl::fromUserInput` said it will attempt to guess the scheme of the input.
It can correctly guess the scheme on linux, but why not on windows?

I feel like a bug in Qt, do you mind add comment in the code, stating that this is a workaround for `QUrl::fromUserInput` ?